### ISSUE TITLE
Relative path in destination vs Linux

### ISSUE THREAD

**Issue by fabh2o** [rclone/rclone#4948]
#### What is the problem you are having with rclone?
Using a relative path (`.` or `..`) in the destination path cause the file (or folder) to be uploaded but with the `.` and `..` not "resolved". That means that on a Linux filesystem those directory become invisible: `rclone mount`, `rclone serve` doesn't show the folder, while `rclone lsjson` (and other `ls` type)  does.

For example:
`rclone copy file GoogleDrive:testDir/./dir1`
`rclone lsjson GoogleDrive:testDir -R | grep file` : `{"Path":"．/dir1/file","Name":"file","Size":32...`

I think that rclone should handle this paths like linux or at least give a warning.

Might be related to [4897](https://github.com/rclone/rclone/issues/4897#issue-777508146)

#### What is your rclone version (output from `rclone version`)
```
rclone v1.53.4
- os/arch: linux/amd64
- go version: go1.15.6
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Manjaro 64bit

####  Which cloud storage system are you using? (e.g. Google Drive)
Google Drive

**Comment by ncw** [comment]
Rclone works like this because on object storage systems, paths with `.` and `..` in are common.

I agree it is a bit odd.

Maybe rclone should optionally normalize the paths on the command line and give a NOTICE if they aren't normalized.

**Comment by fabh2o** [comment]
> Maybe rclone should optionally normalize the paths on the command line and give a NOTICE if they aren't normalized.

If it's a valid thing, should this be labelled as `new feature` or `enhancement`?

**Comment by ivandeex** [comment]
@fabh2o try this
`rclone copy file GoogleDrive:testDir/./dir1 --drive-encoding=None`

**Comment by ivandeex** [comment]
@ncw from https://github.com/rclone/rclone/issues/4897#issuecomment-800557629

> Probably we can introduce an option to normalize paths before operation (with choices `on`,`off`,`warn`) and turn it on by default for local storage.

**Comment by ivandeex** [comment]
This request is similar to #4897 but this one is for cloud storage while #4897 is for local filesystem.

**Comment by fabh2o** [comment]
> @fabh2o try this
> `rclone copy file GoogleDrive:testDir/./dir1 --drive-encoding=None`


Setting the encoding flag has no effect: the folder structure created is the following `testDir/．/dir1/file`
* With the remote mounted, using `ls` and `hexdump` i found out that the "dot" is a ["FULLWIDTH FULL STOP"](https://www.utf8-chartable.de/unicode-utf8-table.pl?start=65280&number=128#:~:text=U%2BFF0E%09%EF%BC%8E%09ef%20bc%208e%09FULLWIDTH%20FULL%20STOP) (the bytes are `ef bc 8e`) and not the "illegal" `.`
* On a web browser, the created folder name seems to be a normal dot
* Using `rclone lsjson GoogleDrive:testDir -R -P=false| grep file | jq '.Path' | hexdump`, the "dot" is the same as the one reported by using rclone mount and ls

**Comment by ivandeex** [comment]
No, it's not a "normal" ASCII dot but a UTF-8 pseudographics glyph that looks like a dot but is treated like a filename character by cloud. rclone performs this and other translations provider-wise.

Normally it should be adjustable via command line by `--BACKEND-encoding=comma,delimited,tokens` (https://rclone.org/overview/#encoding and https://rclone.org/drive/#drive-encoding). I don't understand why it didn't work in your case. Gotta dig around this code https://github.com/rclone/rclone/blob/v1.53.4/backend/drive/drive.go#L491

**Comment by fabh2o** [comment]
I have tried other options for the `--drive-encoding=` flag, the results are in the PS of my previous comment

**Comment by ivandeex** [comment]
Looks like a bug!

**Comment by ivandeex** [comment]
This same trick worked for dropbox - https://github.com/rclone/rclone/issues/4875#issuecomment-800634575

I will investigate why it sucks with drive

**Comment by ivandeex** [comment]
@ncw

Consider https://github.com/rclone/rclone/issues/4897#issuecomment-800557629 and https://github.com/rclone/rclone/issues/4875#issuecomment-800634575

Let's can introduce an option to normalize paths before encoding with choices  `--normalize-path=on,off,warn` (and turn it on by default for local storage).

**Comment by ncw** [comment]
> Let's introduce an option `--normalize-path` to normalize `.` and `..` in paths _before_ encoding
> with choices `always,local,off,warn,?`
> with `local` as default to turn it on by default for local storage only
> ?
> 
> Moreover, it can be put in a _future_ profile #2697

A flag is a good idea.

I wonder whether it might be backend configurable also.

I think all the backends **except** for the bucket based backends (s3, gcs, swift, b2, etc) would always want normalized paths. These backends have a strong concept of a filesystem with folders for which path normalization makes sense.

So maybe the selection should be `always,auto,off,warn` where `auto` looks to see if `f.Features().BucketBased` is set and doesn't normalize paths for those.

What do you think?

> **EDIT**
> If you agree _in general_ to have path normalization in addition to (before / on top) path encoding, let's open another github issue or a forum thread to discuss details. I'm asking _here_ only to stay close to a typical use, for convenience.

OK
